install_deps.R

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env Rscript
+
+# This script installs some development packages that are needed for various
+# apps. It can be sourced from RStudio, or run with Rscript.
+
+
+# Returns the file currently being sourced or run with Rscript
+this_file <- function() {
+  cmdArgs <- commandArgs(trailingOnly = FALSE)
+  needle <- "--file="
+  match <- grep(needle, cmdArgs)
+  if (length(match) > 0) {
+    # Rscript
+    return(normalizePath(sub(needle, "", cmdArgs[match])))
+  } else {
+    # 'source'd via R console
+    return(normalizePath(sys.frames()[[1]]$ofile))
+  }
+}
+
+is_installed <- function (pkg) {
+  if (system.file(package = pkg) == "")
+    FALSE
+  else
+    TRUE
+}
+
+# Install a package or packages if not already installed.
+install_if_needed <- function(pkgs) {
+  installed_idx <- vapply(pkgs, is_installed, TRUE)
+  needed <- pkgs[!installed_idx]
+  if (length(needed) > 0) {
+    message("Installing needed packages from CRAN: ", paste(needed, collapse = ", "))
+    install.packages(needed)
+  }
+}
+
+# Core packages
+install_if_needed(c("devtools", "rsconnect", "packrat", "knitr"))
+
+# Autodetect packages needed for the examples (will install from CRAN)
+install_if_needed(packrat:::dirDependencies(dirname(this_file())))
```

Comment:
Will this fail when trying to install `shinySignals`?

Reply:
It looks like when you call `install.packages()` with multiple packages and one or more isn't available, it installs the available package and just warns about the others.

For example:

```
> install.packages(c('asdlfasdf', 'R6'))
Installing packages into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)
trying URL 'https://cloud.r-project.org/src/contrib/R6_2.4.0.tar.gz'
Content type 'application/x-gzip' length 31545 bytes (30 KB)
==================================================
downloaded 30 KB

* installing *source* package ‘R6’ ...
** package ‘R6’ successfully unpacked and MD5 sums checked
** R
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
* DONE (R6)

The downloaded source packages are in
	‘/tmp/RtmpbOpXST/downloaded_packages’
Warning message:
package ‘asdlfasdf’ is not available (for R version 3.5.3) 
```